woocommerce/woocommerce
# Fix backward compatibility for AdvancedFilters

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




This PR introduces backward compatibility for `AdvancedFilters` and adds a deprecation message. We previously updated the `AdvancedFilters` component to utilize `createInterpolateElement` instead of `interpolateComponents` in [this PR](https://github.com/woocommerce/woocommerce/pull/37967). However, some plugins still send the parameters in the old format to the `AdvancedFilters` component.

With this fix, the `AdvancedFilters` component will handle both cases.

Closes #39021.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install and activate the following plugin [test-advanced-filters-compatibility.zip](https://github.com/woocommerce/woocommerce/files/11926682/test-advanced-filters-compatibility.zip).
2. Go to `WooCommerce` > `Customers` and open the advanced filters.
3. Click on `Add a Filter` and select `Test`.

![Screen Capture on 2023-07-01 at 14-07-43](https://github.com/woocommerce/woocommerce/assets/1314156/7e6e3cc2-8a3b-40b0-9e4b-c825544202b8)

4. The option `Test` is in the old format, but it should still function properly.
5. In the browser console, a deprecation message should appear with the following text:
```
The interpolation string format `{{element}}...{{/element}}` or `{{element/}}` is deprecated. Note: Please update your code to use the new format: `<element>...</element>` or `<element/>`. 
```



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
